### PR TITLE
Feature/initialize database with schema file

### PR DESCRIPTION
**Pull request by KhoiKy**
* When initializing a database, Rivet now runs the migrations found in the schema.ps1 file, which contains the baseline
database schema upon which all migrations should be applied. You can use the `Checkpoint-Migration` function to create
a baseline `schema.ps1` file for your database(s).

### CODE REVIEW

File: Rivet/Functions/Checkpoint-Migration.ps1 [hunk 1]
```diff
@@ -66,14 +74,18 @@ function Checkpoint-Migration
...
-        Set-Content -Path $OutputPath -Value $migration
+        $exportedMigration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
+        $exportedMigration = $exportedMigration -join [Environment]::NewLine
 
-        foreach( $migration in $pushedMigrations )
+        # Now pop those pushed migrations and remove their respective migration files.
+        foreach( $pushedMigration in $pushedMigrations )
         {
-            $migrationFilePath = Join-Path -Path $databaseItem.MigrationsRoot -ChildPath "$($migration.MigrationFileName).ps1"
+            Invoke-Rivet -Pop -Database $databaseItem.Name -Environment $Environment -ConfigFilePath $ConfigFilePath -Name $pushedMigration.MigrationFileName
```
Comment (splatteredbits) [inline Rivet/Functions/Checkpoint-Migration.ps1]: Don't pop any existing migrations. Their changes will be removed from the database and then won't get captured in the schema export.

File: Rivet/Functions/Get-MigrationFile.ps1 [hunk 2]
```diff
@@ -56,22 +56,44 @@ function Get-MigrationFile
             Write-Debug -Message $_ 
             if( (Test-Path -Path $_ -PathType Container) )
             {
-                Get-ChildItem -Path $_ -Filter '*_*.ps1'
+                Get-ChildItem -Path $_ -Filter '*.ps1'
```
Comment (splatteredbits) [inline Rivet/Functions/Get-MigrationFile.ps1]: This should stay `*_*.ps1`. Why did you change it?

File: Rivet/Functions/Get-MigrationFile.ps1 [hunk 3]
```diff
@@ -56,22 +56,44 @@ function Get-MigrationFile
...
             elseif( (Test-Path -Path $_ -PathType Leaf) )
             {
                 Get-Item -Path $_
             }
         } | 
         ForEach-Object {
-            if( $_.BaseName -notmatch '^(\d{14})_(.+)' )
+            if( $_.BaseName -eq 'schema' )
+            {
+                $schemaContents = Get-Content -Path $_.FullName
```
Comment (splatteredbits) [inline Rivet/Functions/Get-MigrationFile.ps1]: This should be reverted to what you had before. `Update-Database` will pass in the path to the schema.ps1 file to run. All you need to do here is detect that it's the schema.ps1 file and set the id/name.

File: Rivet/Functions/Update-Database.ps1 [hunk 4]
```diff
@@ -142,14 +143,14 @@ function Update-Database
         Get-MigrationFile -Path $Path -Configuration $Configuration @byName -ErrorAction Stop |
         Sort-Object -Property 'MigrationID' -Descending:$popping |
         Where-Object {
-            if( $RivetSchema )
+            if( $RivetSchema -or ($_.BaseName -eq 'schema') )
```
Comment (splatteredbits) [inline Rivet/Functions/Update-Database.ps1]: Just use `if ($RivetSchema)`.

File: Test/RivetTest/Functions/Clear-TestDatabase.ps1 [hunk 5]
```diff
@@ -22,7 +22,7 @@ function Clear-TestDatabase
 
     if( Test-Database -Name $Name )
     {
-        $query = "select * from [$($Name)].[rivet].[Migrations] where ID > 1000000000 order by ID"
+        $query = "select * from [$($Name)].[rivet].[Migrations] where ID > 10101000000 order by ID"
```
Comment (splatteredbits) [inline Test/RivetTest/Functions/Clear-TestDatabase.ps1]: Use a module-level variable for this magic number and update the other places that use it to use the variable.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: Rivet/Functions/Get-MigrationFile.ps1 (hunk 3 above)
Comment (KhoiKy) [reply Rivet/Functions/Get-MigrationFile.ps1]: If I don't compare the contents of the schema file here, and the schema file happens to be only the default contents. Convert-FileToMigration will throw errors on how the migrations doesn't have any push or pop operations.

File: Rivet/Functions/Update-Database.ps1 (hunk 4 above)
Comment (KhoiKy) [reply Rivet/Functions/Update-Database.ps1]: If I don't check for the schema file here, it will throw an error in the next check `if( [int64]$_.MigrationID -lt $script:firstMigrationId )` and the schema file ID would be less than the first migration ID, throwing an error.

File: Rivet/Functions/Checkpoint-Migration.ps1 (hunk 1 above)
Comment (KhoiKy) [reply Rivet/Functions/Checkpoint-Migration.ps1]: The export happens on line 78 before the Pop happens.

File: Rivet/Functions/Get-MigrationFile.ps1 (hunk 2 above)
Comment (KhoiKy) [reply Rivet/Functions/Get-MigrationFile.ps1]: It doesn't pull the schema.ps1 file as it doesn't fit the naming convention for that filter.

File: Rivet/Functions/Get-MigrationFile.ps1 (hunk 3 above)
Comment (splatteredbits) [reply Rivet/Functions/Get-MigrationFile.ps1]: Update `Convert-FileToMigration` to no longer write this error.

File: Rivet/Functions/Update-Database.ps1 (hunk 4 above)
Comment (splatteredbits) [reply Rivet/Functions/Update-Database.ps1]: Using `$RivetSchema` should skip that check.

Comment (splatteredbits) [reply Rivet/Functions/Update-Database.ps1]: `Initialize-Database` calls `Update-Database` with the `-RivetSchema` switch when pushing the schema.ps1 file, so `if ($RivetSchema)` should return true and you should never get to the check on line 151.

File: Rivet/Functions/Checkpoint-Migration.ps1 (hunk 1 above)
Comment (splatteredbits) [reply Rivet/Functions/Checkpoint-Migration.ps1]: Regardless, don't pop any migrations. Just export the current schema and delete all applied migrations.

File: Rivet/Functions/Get-MigrationFile.ps1 (hunk 2 above)
Comment (splatteredbits) [reply Rivet/Functions/Get-MigrationFile.ps1]: The check on line 57 should be false because $_ is a file (the schema.ps1 file), so it should flow down to the check on line 61, which checks if the item is a file. The schema.ps1 file is, so the file should be returned by `Get-Item`.

File: Rivet/Functions/New-Migration.ps1 [hunk 6]
```diff
@@ -38,6 +41,15 @@ function New-Migration
         $migrationPath = [IO.Path]::GetFullPath( $migrationPath )
         New-Item -Path $migrationPath -Force -ItemType File
 
+        $schemaPs1Path = Join-Path -Path $Path -ChildPath $script:schemaFileName
+        if (-not (Test-Path -Path $schemaPs1Path))
+        {
+            $script:defaultSchemaPs1Content | Set-Content -Path $schemaPs1Path
+            $msg = "Rivet created ""$($schemaPs1Path | Resolve-Path -Relative)"", a file where Rivet to stores the " +
```
Comment (splatteredbits) [inline Rivet/Functions/New-Migration.ps1]: Change "Rivet to stores" to "Rivet stores".

**Review by splatteredbits — APPROVED** [review]
